core/dive.c

```diff
@@ -747,7 +747,7 @@ void copy_samples(struct divecomputer *s, struct divecomputer *d)
 void alloc_samples(struct divecomputer *dc, int num)
 {
 	if (num > dc->alloc_samples) {
-		dc->alloc_samples = (dc->alloc_samples * 3) / 2 + 10;
+		dc->alloc_samples = (num * 3) / 2 + 10;
```

Comment:
So this allocates 50% + 10 additional samples. Which of course works fine if your pattern everywhere in the code is to either simply allocate "one more" sample (so this gets called with 'nr+1'), or up to six samples in one block.
That is of course bleeping fragile and needs to be fixed - but given our call pattern, I wonder how you got into trouble here?

Reply:
just for the record, this did not bring trouble, but I saw it while debugging, and thought it looked wrong. I did not adjust the algorithm as such, the structure is not very big so overallocating is not a big issue.